tools/src/file_format/analysis.rs

```diff
@@ -103,6 +111,7 @@ pub struct AnalysisSource {
     pub pretty: String,
     pub sym: Vec<String>,
     pub no_crossref: bool,
+    pub nesting_range: SourceRange,
 }
 
 impl AnalysisSource {
```

Comment:
The `AnalysisSource::merge` function I think also needs to be updated so that it unions the nesting_range of `other` into `self`. Consider for example c++ code like so:
```c++
void func()
#ifdef LINUX
{ linux_body; }
#else 
{ other_body; }
#endif 
```

In this case, we'll get the analysis data output by clang for two different platforms, which will have different nesting ranges for the symbol, and then we'll run through this merge code to merge those source entries. Really we want to treat the union of the different function bodies as being the impl of `func`.